src/UserList.tsx

```diff
@@ -23,47 +23,48 @@ const Card = styled.a`
   ${justifyContent}
 `;
 
-type Props = {
-  query:  UserList_query
-}
-class UserList extends React.Component<Props> {
-  render() {
-    const { query } = this.props;
-    const { users } = query;
-
-    return (
-      <Flex flexDirection='column'>
-        {users.edges.map(({node}) => (
-          <Card key={node.id}>
-            <span>User: {node.name}</span>
-            <span>Email: {node.name}</span>
-          </Card>
-        ))}
-      </Flex>
-    )
-  }
-}
+const UserList = () => {
+  const response = useLazyLoadQuery<AppQuery>(
+    graphql`
+      query UserListQuery {
+        ...UserList_query
+      }
+    `,
+    { },
+    {
+      fetchPolicy: 'network-only',
+    },
+  );
```

Comment:
One approach we have at work is to only declare fragments in components and recieve the fragement ref as props and have queries outside(as wrappers if needed to be isolated), so that way components are not attached to the query it self and can be easily reused. Whats you thoughts on that?

Reply:
Usually you only have fragments at the component

but if the component is the top level component of a route, it can also have the query itself, so you have less components in your tree

if you need to reuse top level components, you can also split later on

Reply:
Yeah, thats the pattern that we follow 👍